### DISCUSSION TITLE
more flexible `goto_file`

### DISCUSSION THREAD

**Discussion by ModProg** [helix-editor/helix#10213] [Ideas]
After opening #8822 I noticed that there are more things I miss from `vim` in regard to `gf`.

1. Allow non `.` punctuation after a path, e.g., `path,` or `"path";` should open `path` and not `path,` and `path";` respectively, if you want the latter behavior it is easy enough to just do a `f,gf`.  (maybe we could have a config option for what characters are path, like `isfname` in VIM)
2. Support paths that omit the file extension. Simple behavior could be “if file not exist, try appending current extension”, but a more flexible tool could be a file type based `suffixesadd` (VIM) like list of extensions to try.
3. VIM also as a `includeexpr` which allows transforming a path by replacing characters, i.e., `.` with `/` in java imports. (This is applied after the normal behavior failed.) This might be best implemented when the scripting support lands, though we could also just have a `regex, replacements` tuple that would probably already cover most things.

Not every file has an LSP, and this just allows navigating these almost as if one had one.

**Comment by ModProg** [comment]
I feel like at least 1. is very actionable, and I'd look into implementing this and #8822. 

If you agree with the other additions, I'd look into those as well, though I'd need some input on how the config options should look.

**Comment by daedroza** [comment]
I am unsure about the direction of VIM about such this feature but can we instead have file picker show up with the relevant text in search box?

Until it is implemented is there a way to map a new keyboard shortcut to search for file using file picker with selected text under cursor?

**Comment by matoous** [comment]
Personally, I feel like points 2. and 3. would be better left for plugin (once that's possible). Point 1. seems like a reasonable requirement that would ideally be fixed.

**Comment by TornaxO7** [comment]
for anyone who subscribed to this: Could you try out https://github.com/helix-editor/helix/pull/9065 pls? `gf` should now basically behave similar to `gf` in neovim.
